dbms/src/Flash/Mpp/ExchangeReceiver.cpp

```diff
@@ -471,7 +475,21 @@ ExchangeReceiverBase<RPCContext>::~ExchangeReceiverBase()
 template <typename RPCContext>
 void ExchangeReceiverBase<RPCContext>::cancel()
 {
-    setEndState(ExchangeReceiverState::CANCELED);
+    if (setEndState(ExchangeReceiverState::CANCELED))
+    {
+        if (isReceiverForTiFlashStorage())
+        {
+            try
+            {
+                rpc_context->cancelMPPTaskOnTiFlashStorageNode(exc_log);
+            }
+            catch (...)
```

Comment:
Can we move the error handling to `cancelMPPTaskOnTiFlashStorageNode`?

Reply:
OK.